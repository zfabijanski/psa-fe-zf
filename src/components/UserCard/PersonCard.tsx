import { FormattedMessage } from "react-intl";
import styled from "styled-components";
import profilePlacePlaceholder from "../../assets/newIcons/avatar.svg";

const ProfilePhotoDiv = styled.div`
  width: 100px;
  height: 100px;
  margin: 48px auto 24px;
  background-color: ${({ theme }) => theme.newColors.white100};
  border-radius: 50%;
`;

const ProfilePhotoImg = styled.img`
  border-radius: 50%;
  width: 100px;
  border: 4px solid ${({ theme }) => theme.newColors.primary100};
`;

const ConsultantFullname = styled.p`
  font-size: 24px;
  font-weight: 600;
  font-stretch: normal;
  line-height: 28px;
  text-align: center;
  color: ${({ theme }) => theme.newColors.gray100};
  margin-bottom: 0;
`;

const ConsultantTitle = styled.p`
  font-size: 14px;
  font-stretch: normal;
  line-height: 20px;
  text-align: center;
  color: ${({ theme }) => theme.newColors.primary100};
  margin-top: 4px;
  white-space: pre-wrap;
`;

export interface IPersonCard {
  fullName: string;
  image?: string;
  title: string;
}

export const PersonCard = (props: IPersonCard) => (
  <div>
    <ProfilePhotoDiv>
      {props.image ? (
        <ProfilePhotoImg src={props.image} alt="profile-avatar" />
      ) : (
        <img src={profilePlacePlaceholder} alt="profile-avatar-placeholder" />
      )}
    </ProfilePhotoDiv>
    <ConsultantFullname>{props.fullName}</ConsultantFullname>
    <ConsultantTitle>
      <FormattedMessage id={props.title} />
    </ConsultantTitle>
  </div>
);

export default PersonCard;
