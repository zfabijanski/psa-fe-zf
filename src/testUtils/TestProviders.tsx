import { render, RenderOptions } from "@testing-library/react";
import { FC, PropsWithChildren, ReactElement } from "react";
import { Provider } from "react-redux";
import { ThemeProvider } from "styled-components";
import { Normalize } from "styled-normalize";
import GlobalStyles from "theme/globalStyles";
import { theme } from "theme/theme";
import rootStore, { configureStore, rootReducer as reducer } from "AppStore";
import { BrowserRouter } from "react-router-dom";
import { Modal } from "components/UI/Modal";
import { IntlProvider } from "react-intl";
import { plMessages } from "translations/pl";
import I18nProvider from "I18nProvider";

interface TestProvidersProps {
  store?: typeof rootStore;
  renderModal?: boolean;
  realIntlProvider?: boolean;
}

export const addTestProviders =
  ({
    store = configureStore({ reducer }),
    renderModal = true,
    // @TODO: update tests to use realIntlProvider everywhere
    realIntlProvider = false,
  }: TestProvidersProps): FC<PropsWithChildren<{}>> =>
  ({ children }) => {
    const TranslationsProvider = realIntlProvider ? IntlProvider : I18nProvider;

    return (
      <Provider store={store}>
        <BrowserRouter>
          <ThemeProvider theme={theme}>
            <TranslationsProvider locale="pl" messages={plMessages}>
              <Normalize />
              <GlobalStyles />
              {children}
              {renderModal && <Modal />}
            </TranslationsProvider>
          </ThemeProvider>
        </BrowserRouter>
      </Provider>
    );
  };

const customRender = (
  ui: ReactElement,
  options: Omit<RenderOptions, "options"> = {},
  testProviderOptions: TestProvidersProps = {}
) => render(ui, { wrapper: addTestProviders(testProviderOptions), ...options });

export * from "@testing-library/react";

// override render method
export { customRender as render };
