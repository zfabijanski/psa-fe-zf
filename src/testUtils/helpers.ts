export const sleep = (ms: number = 100): Promise<unknown> =>
  new Promise((resolve) => setTimeout(resolve, ms));
