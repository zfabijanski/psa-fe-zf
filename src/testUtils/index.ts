export * from "./TestProviders";
export * from "./mockCreators";
export * from "./helpers";
