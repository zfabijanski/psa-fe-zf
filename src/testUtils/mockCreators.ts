import {
  ICalculatorCoversConfig,
  IProductCovers,
} from "features/Calculator/types";
import {
  DocumentState,
  IDirectory,
  IDirectoryMetadata,
  IDocumentMetadata,
} from "features/Library/types";
import { IMeeting } from "features/Meeting/History/types";
import { IIllustration, ProductGuid } from "features/Products/types";
import { IModalData, ModalTypes } from "models/Modal";
import {
  Frequency,
  IFundRiskProfileConfig,
  IProductConfig,
  IProductFund,
  IProductGuarantee,
  InvestmentProfile,
  Role,
} from "models/common";
import {
  CalculationStatus,
  Direction,
  ICalculationDetailsDto,
  ICover,
  IInsured,
  YesNo,
} from "models/calculator";

export const createCalculationDetailsDto = (
  overrides?: Partial<ICalculationDetailsDto>
): ICalculationDetailsDto => ({
  calculation_id: 1,
  working_date: "2020-01-01",
  premiums_total: 100,
  guaranteed_death_benefit: 100,
  guaranteed_maturity_benefit: 100,
  projected_maturity_value_egr_m: 100,
  product_categories: [],
  uuid: "uuid",
  version: "version",
  source_system: "source_system",
  product_guid: 1,
  insureds_list: [],
  covers_list: [],
  funds_list: [],
  ...overrides,
});

export const createInsured = (overrides?: Partial<IInsured>): IInsured => ({
  insured_idx: 1,
  insured_uuid: "insured_uuid",
  age_in_months: 1,
  date_of_birth: "2020-01-01",
  role: Role.MAIN,
  ...overrides,
});

export const createCover = (overrides?: Partial<ICover>): ICover => ({
  cover_id: 1,
  cover_idx: 1,
  insured_idx: 1,
  code: "code",
  frequency_code: Frequency.Annually,
  payments: 1,
  direction_of_calculation: Direction.SumAssuredToPremium,
  investment_profile: InvestmentProfile.Balanced,
  indexation: YesNo.Yes,
  sum_assured: 1,
  premium_per_freq: 1,
  duration: 1,
  idx_option: YesNo.Yes,
  rop_option: YesNo.Yes,
  add_sum_assured: 1,
  ending_age: 1,
  exp_mat_ben: 1,
  ...overrides,
});

export const createCalculatorCoversConfig = (
  overrides?: Partial<ICalculatorCoversConfig>
): ICalculatorCoversConfig => ({
  coverName: "coverName",
  isChild: false,
  isHospital: false,
  ...overrides,
});

export const createCalculatorCoversConfigMap = (
  overrides?: Partial<Record<string, ICalculatorCoversConfig>>
): Record<string, ICalculatorCoversConfig> => ({
  coverName1: createCalculatorCoversConfig({ coverName: "coverName1" }),
  coverName2: createCalculatorCoversConfig({ coverName: "coverName2" }),
  ...overrides,
});

export const createProductCovers = (
  overrides?: Partial<IProductCovers>
): IProductCovers => ({
  mainCover: "mainCover",
  additionalCovers: ["additionalCover1", "additionalCover2"],
  wopCovers: ["wopCover1", "wopCover2"],
  childCovers: ["childCover1", "childCover2"],
  ...overrides,
});

export const createProductCoversMap = (
  overrides?: Partial<Record<string, IProductCovers>>
): Record<string, IProductCovers> => ({
  mainCover1: createProductCovers({ mainCover: "mainCover1" }),
  mainCover2: createProductCovers({ mainCover: "mainCover2" }),
  ...overrides,
});

export const createProductConfig = (
  overrides?: Partial<IProductConfig>
): IProductConfig => ({
  product_guid: 1,
  product_name: "product_name",
  product_categories: [],
  product_freqs: [],
  product_covers: [],
  product_funds: [],
  ...overrides,
});

export const createProductConfigMap = (
  overrides?: Partial<Record<string, IProductConfig>>
): Record<string, IProductConfig> => ({
  1: createProductConfig({ product_name: "product_name1" }),
  2: createProductConfig({ product_name: "product_name2" }),
  ...overrides,
});

export const createFundRiskProfileConfig = (
  overrides?: Partial<IFundRiskProfileConfig>
): IFundRiskProfileConfig => ({
  code: "code",
  name_pl: "name_pl",
  name_en: "name_en",
  risk_level: 1,
  status_code: "status_code",
  notes: "notes",
  ...overrides,
});

export const createProductGuarantee = (
  overrides?: Partial<IProductGuarantee>
): IProductGuarantee => ({
  guarantee_code: "guarantee_code",
  guarantee_name: "guarantee_name",
  guarantee_min_value: "1",
  guarantee_duration_months: "4",
  guarantee_percentages: [{ guarantee_percent: 10 }],
  ...overrides,
});

export const createProductFund = (
  overrides?: Partial<IProductFund>
): IProductFund => ({
  cover_code: "cover_code",
  fund_code: "fund_code",
  fund_name: "fund_name",
  fund_risk_profile: "code",
  fund_order: 1,
  guarantees: [createProductGuarantee()],
  ...overrides,
});

export const createModalData = (
  overrides: Partial<IModalData> = {}
): IModalData => ({
  modalIllustration: "alert-fill",
  modalTitleTrKey: "modalTitleTrKey",
  modalContentTrKey: "modalContentTrKey",
  modalContent: "modalContent",
  modalButtons: {
    confirm: { textTrKey: "modal.ok" },
  },
  modalType: ModalTypes.default,
  maskClosable: true,
  ...overrides,
});

export const createIllustration = (
  overrides: Partial<IIllustration> = {}
): IIllustration => ({
  idd_id: 1,
  product_guid: ProductGuid.Savings,
  adequacy_id: 1,
  order_id: 1,
  expected_amount: { value: 3 },
  premium_total: { value: 3 },
  duration: { value: 3 },
  premium_main_cover: { value: 3 },
  frequency: { value: 3 },
  sum_of_premiums: { value: 3 },
  sum_assured: { value: 3 },
  investment_profile: { value: 3 },
  pru_calc_calculation_id: 1,
  status: CalculationStatus.Illustrated,
  json_needs_analysis: null,
  ...overrides,
});

export const createDirectory = (
  overrides?: Partial<IDirectory>
): IDirectory => ({
  directoryId: 1,
  directories: [],
  documentsIds: [1, 2, 3],
  ...overrides,
});

export const createDirectoryMetadata = (
  overrides?: Partial<IDirectoryMetadata>
): IDirectoryMetadata => ({
  directoryId: 1,
  name: "name",
  isOpen: false,
  ...overrides,
});

export const createDocumentMetadata = (
  overrides?: Partial<IDocumentMetadata>
): IDocumentMetadata => ({
  documentId: 1,
  guid: "guid",
  name: "name",
  state: DocumentState.NotAttached,
  isStateChanging: false,
  moduleName: "PRU_LIBRARY",
  ...overrides,
});

export const createMeeting = (overrides?: Partial<IMeeting>): IMeeting => ({
  meetingNo: "meetingNo",
  meetingId: 1,
  name: "name",
  age: 18,
  lastAccessDate: new Date("2023-01-01"),
  createDate: new Date("2023-01-01"),
  statusId: 1,
  statusName: "statusName",
  statusDescription: "statusDescription",
  state: "state",
  haveAdequacy: false,
  haveCalculation: false,
  haveIdd: false,
});
