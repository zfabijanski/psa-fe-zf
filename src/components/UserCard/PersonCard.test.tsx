import { PersonCard, IPersonCard } from "./PersonCard";
import { render, screen } from "testUtils";

const defaultProps: IPersonCard = {
  fullName: "John Doe",
  image: "https://via.placeholder.com/100",
  title: "Consultant",
};

const setup = (props: Partial<IPersonCard> = {}) =>
  render(<PersonCard {...defaultProps} {...props} />);

describe("PersonCard", () => {
  it("should render correctly", () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  it("should render the correct name", () => {
    setup();
    expect(screen.getByText("John Doe")).toBeInTheDocument();
  });

  it("should render the correct title", () => {
    setup();
    expect(screen.getByText("Consultant")).toBeInTheDocument();
  });

  it("should render the correct image", () => {
    setup();
    expect(screen.getByAltText("profile-avatar")).toBeInTheDocument();
  });

  it("should render the correct image when image is not provided", () => {
    setup({ image: undefined });
    expect(
      screen.getByAltText("profile-avatar-placeholder")
    ).toBeInTheDocument();
  });
});
