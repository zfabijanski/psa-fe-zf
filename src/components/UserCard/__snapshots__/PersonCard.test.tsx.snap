// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonCard should render correctly 1`] = `
.c0 {
  width: 100px;
  height: 100px;
  margin: 48px auto 24px;
  background-color: #FFFFFF;
  border-radius: 50%;
}

.c1 {
  border-radius: 50%;
  width: 100px;
  border: 4px solid #055A60;
}

.c2 {
  font-size: 24px;
  font-weight: 600;
  font-stretch: normal;
  line-height: 28px;
  text-align: center;
  color: #2F2F2F;
  margin-bottom: 0;
}

.c3 {
  font-size: 14px;
  font-stretch: normal;
  line-height: 20px;
  text-align: center;
  color: #055A60;
  margin-top: 4px;
  white-space: pre-wrap;
}

<div>
  <div>
    <div
      class="c0"
    >
      <img
        alt="profile-avatar"
        class="c1"
        src="https://via.placeholder.com/100"
      />
    </div>
    <p
      class="c2"
    >
      John Doe
    </p>
    <p
      class="c3"
    >
      Consultant
    </p>
  </div>
  <div />
</div>
`;
